### DISCUSSION TITLE
How to escape double quote

### DISCUSSION THREAD

**Discussion by tsawyer999** [xunit/xunit#2583] [Question] [answered]
Hello,

I am trying to display the name for a test:
> Parameter "RETURN VALUE" is always added

I tried those two ways and was not able to accomplish the expected result.

>        [Fact]
>        public void Parameter_X22RETURN_VALUEX22_is_always_added()

Result: 
> Parameter "RETURN VALUEX22 is always added

>        [Fact]
>        public void Parameter_X22RETURN_VALUE_X22_is_always_added()

Result: 
>Parameter "RETURN VALUE_" is always added

Can someone give me a hand to achieve the desired result?

Thank you very much!

**Comment by bradwilson** [comment, marked as answer]
Yep, this is a bug, and a rather sneaky one at that.

If your method were called:

`public void Parameter_X22RETURN_VALX22_is_always_added()`

Then it shows up as expected:

`Parameter "RETURN VAL" is always added`

The problem is the parsing code sees the `U` in `VALUE` and tries to look at the next four characters to see if they're hex digits. It sees the `E`, which is legal, and keeps going; it sees the `X`, which is not, and then stops.

Unfortunately, what it's done at this point is incorrect: it puts the literal `UEX` into the output and then moves on looking for something else.

This is why adding a _second_ illegal character before the `X` causes the `X` to be correctly interpreted, because the buggy code consumes the `UE_` and then fails, and then goes back looking for new things. In reality, what it should do is only consume the `U` and then go back for and look again.

The only work around unfortunately is what you've already determined: the insertion of an extra non-hex character that the algorithm can consume safely without landing on and consuming the `X`, which then breaks the hex encoding. 😞 

I'll open an issue for this.

**Reply by bradwilson** [reply]
Issue #2651

**Comment by tsawyer999** [comment]
Ho Ho Brad! Thank you very much for your dedication! We will use that work-around!
